appenders/async/src/append.rs

```diff
@@ -56,11 +56,19 @@ impl Append for Async {
     }
 
     fn flush(&self) -> Result<(), Error> {
-        let overflow = self.overflow;
+        let (completion, done) = crossbeam_channel::bounded(0);
         let task = Task::Flush {
             appends: self.appends.clone(),
+            completion,
         };
-        self.state.send_task(task, overflow)
+        self.state.send_task(task, Overflow::Block)?;
+        match done.recv() {
+            Ok(Ok(())) => Ok(()),
+            Ok(Err(err)) => Err(err),
+            Err(_) => Err(Error::new(
+                "async appender worker exited before completing flush",
+            )),
+        }
```

Comment:
I'm not sure if I missed some context, but isn't the point of async appender not to block the current thread? If we wait for the flushing result here wouldn't it be no different than sync appender?